### PR TITLE
AP_ESC_Telem: log NaNs on no measurements, various cleanups

### PR DESCRIPTION
**Pull request by mbuzdalov**
This PR is, in a sense, continuation of #25978 as, among other things, it implements a suggestion from that conversation.

What it does is:

- initializes RPM values, to be subsequently logged, with NaN instead of 0, so that an absence of a measurement will be reported as such;
- in `update()`, deduplicates calls to `micros64()` when actually logging ESC data;
- replaces repeated constructions such as `_telem_data[esc_index]` with an appropriate reference;
- cleans up some whitespace and redundant initialization.

### CODE REVIEW

File: libraries/AP_ESC_Telem/AP_ESC_Telem.cpp [hunk 1]
```diff
@@ -325,7 +331,7 @@ void AP_ESC_Telem::send_esc_telemetry_mavlink(uint8_t mav_chan)
         }
 
         bool all_stale = true;
-        for (uint8_t j=0; j<4; j++) {
+        for (uint8_t j = 0; j < 4; j++) {
```
Comment (peterbarker) [inline libraries/AP_ESC_Telem/AP_ESC_Telem.cpp:334]: This is contrary to most of our code.

File: libraries/AP_ESC_Telem/AP_ESC_Telem.cpp [hunk 2]
```diff
@@ -425,41 +432,42 @@ void AP_ESC_Telem::update_telem_data(const uint8_t esc_index, const AP_ESC_Telem
...
     }
     if (data_mask & AP_ESC_Telem_Backend::TelemetryType::USAGE) {
-        _telem_data[esc_index].usage_s = new_data.usage_s;
+        telemdata.usage_s = new_data.usage_s;
     }
 
-    _telem_data[esc_index].count++;
-    _telem_data[esc_index].types |= data_mask;
-    _telem_data[esc_index].last_update_ms = AP_HAL::millis();
+    ++telemdata.count;
```
Comment (peterbarker) [inline libraries/AP_ESC_Telem/AP_ESC_Telem.cpp:468]: This is also contrary to the vast majority of our code.

**Review by peterbarker — CHANGES_REQUESTED** [review]
This is a series of really nice cleanups, and I really quite like it.

Something I particularly like about it is that it might help clean up this horrendous interface - there are no bounds checks being done in here, meaning if the caller screws up we could end up crashing.

Just a couple of little stylistic things to change.

**Comment by amilcarlucas** [comment]
```
Binary Name      Text [B]       Data [B]     BSS (B)      Total Flash Change [B] (%)      Flash Free After PR (B)
---------------  -------------  -----------  -----------  ----------------------------  -------------------------
ardurover        -8 (-0.0005%)  0 (0.0000%)  0 (0.0000%)  -8 (-0.0005%)                                    337[28](https://github.com/ArduPilot/ardupilot/actions/runs/7949731710/job/21701231777?pr=26252#step:10:29)4
blimp            -8 (-0.0006%)  0 (0.0000%)  0 (0.0000%)  -8 (-0.0006%)                                    623024
arducopter       -8 (-0.0005%)  0 (0.0000%)  0 (0.0000%)  -8 (-0.0005%)                                    19[29](https://github.com/ArduPilot/ardupilot/actions/runs/7949731710/job/21701231777?pr=26252#step:10:30)[32](https://github.com/ArduPilot/ardupilot/actions/runs/7949731710/job/21701231777?pr=26252#step:10:33)
arduplane        -8 (-0.0005%)  0 (0.0000%)  0 (0.0000%)  -8 (-0.0005%)                                    201240
ardusub          -8 (-0.0005%)  0 (0.0000%)  0 (0.0000%)  -8 (-0.0005%)                                    400560
antennatracker   -8 (-0.0006%)  0 (0.0000%)  0 (0.0000%)  -8 (-0.0006%)                                    645892
arducopter-heli  -8 (-0.0005%)  0 (0.0000%)  0 (0.0000%)  -8 (-0.0004%)                                    186764
```

File: libraries/AP_ESC_Telem/AP_ESC_Telem.cpp (hunk 1 above)
Comment (mbuzdalov) [reply libraries/AP_ESC_Telem/AP_ESC_Telem.cpp:334]: I see only that there is no consistency regarding spacing in the loops, and if there is a trend, the older code tens to use no spaces, and the newer code tends to use spaces.

- In the (https://ardupilot.org/dev/docs/style-guide.html)[Style Guide], there is nothing about loops, and the only loop example has spaces.
- In this file, loops with spaces are more common (this one was the only loop without spaces).
- `RC_Channel.cpp` has four loops without spaces and one foreach loop with spaces.
- `AP_Scripting.cpp` has three loops without spaces and two loops (one of them foreach) with spaces.
- `AP_Scripting_CANSensor.cpp` has one loop, and it is with spaces.
- `AP_RCProtocol_CRSF.cpp` has five loops, all with spaces.
- `AP_Vehicle.cpp` has three loops with spaces and two "hybrid" loops with no space in the comparison.

File: libraries/AP_ESC_Telem/AP_ESC_Telem.cpp (hunk 2 above)
Comment (mbuzdalov) [reply libraries/AP_ESC_Telem/AP_ESC_Telem.cpp:468]: Here I agree, I will fix this in the next force push.

File: libraries/AP_ESC_Telem/AP_ESC_Telem.cpp (hunk 1 above)
Comment (mbuzdalov) [reply libraries/AP_ESC_Telem/AP_ESC_Telem.cpp]: Okay, will leave this one as is, new version is force-pushed now.

**Review by andyp1per — APPROVED** [review]
LGTM

**Review by peterbarker — APPROVED** [review]

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**